### ISSUE TITLE
Urdu font not rendering correctly

### ISSUE THREAD

**Issue by TheBluuDot** [mpv-player/mpv#9977]
### Important Information

Provide following Information:
- mpv version: 0.34.0-221-g1c49d5735d 
- Platform and Version: Edition	Windows 10 Pro 21H2 19044.1526
- Source of the mpv binary: shinchiro

### Bug

Pay attention to this dot ![image](https://user-images.githubusercontent.com/62665768/158040976-a8f25c8c-93ce-4867-b80a-bc789479605a.png) under the letter. It renders fine in other libass implementation like AegiSubDC or vlc. Only mpv seems to have this issue.

### Reproduction steps

Put the font in `mpv/config/fonts`
Play any video in mpv with --sub-font Jameel Noori Nastaleeq and the attached subtitles 
The bug in question is visible from `00:01:18.86` to `00:01:24.05`, `00:02:08.50` to `00:02:13.73`and `00:04:00.72` to `00:04:05.95`

### Expected behavior
![image](https://user-images.githubusercontent.com/62665768/158040835-db201aab-6896-428b-ab59-0b1214620516.png)

### Actual behavior
![image](https://user-images.githubusercontent.com/62665768/158040165-bfa3a38a-6bda-4564-a03e-963aeec030e4.png)

### Log file

[log.txt](https://github.com/mpv-player/mpv/files/8238783/log.txt)

### Sample files
[Urdu Font](https://github.com/fireworkstech/UrduFontsForWeb/raw/master/Jameel%20Noori%20Nastaleeq.ttf)
[UrduSubtitles.zip](https://github.com/mpv-player/mpv/files/8238767/UrduSubtitles.zip)

**Comment by TheBluuDot** [comment]
So I checked the previous versions and this issue popped up somewhere between [fd63bf3](https://github.com/mpv-player/mpv/commit/fd63bf398af14c95ee7df18815dc3c9d1b4d9b49) and [d92cf77](https://github.com/mpv-player/mpv/commit/d92cf77be56190dfd7fbac3724f290ceb6ff9437).

**Comment by TheOneric** [comment]
This is caused by a change in HarfBuzz, specifically since https://github.com/harfbuzz/harfbuzz/commit/43be5ba442548528c89ad31c0927cc68515b736e.

If you believe the older rendering is the correct one, then please file an issue with HarfBuzz providing the font, the two images and the `میرے ڈھول جدائیاں دی تینوں خبر کیویں ہووے` sample string. To make it easier for the HarfBuzz devs, you can also point out that the difference shows up with their `hb-shape` utility.
```sh
# With prior versions
$ util/hb-shape "Jameel Noori Nastaleeq.ttf" 'میرے ڈھول جدائیاں دی تینوں خبر کیویں ہووے'
[.notdef280=40+1873|.notdef251=39+561|.notdef18216=37+1183|.notdef293=36+5|.notdef19365=34+1470|.notdef12797=31+1366|.notdef293=30+5|.notdef8030=27+1556|.notdef293=26+5|.notdef276=25+1228|.notdef1650=21+0|.notdef465=21@390,1635+0|.notdef459=21@630,375+0|.notdef467=21@1200,1815+0|.notdef805=21+1455|.notdef293=20+5|.notdef279=19+1116|.notdef233=18+571|.notdef293=17+5|.notdef276=16+1228|.notdef18694=13+1032|.notdef225=12+299|.notdef1651=10+0|.notdef457=10@600,75+0|.notdef1547=10+1375|.notdef293=9+5|.notdef247=8+1169|.notdef18504=6+1251|.notdef271=5+591|.notdef293=4+5|.notdef280=3+1873|.notdef16651=0+1619]

# Since 43be5ba and also on current master 49fb8f9072
$ util/hb-shape "Jameel Noori Nastaleeq.ttf" 'میرے ڈھول جدائیاں دی تینوں خبر کیویں ہووے'
[.notdef280=40+1873|.notdef251=39+561|.notdef18216=37+1183|.notdef293=36+5|.notdef19365=34+1470|.notdef12797=31+1366|.notdef293=30+5|.notdef8030=27+1556|.notdef293=26+5|.notdef276=25+1228|.notdef1650=21+0|.notdef465=21@390,1635+0|.notdef459=21@630,375+0|.notdef467=21@1200,1815+0|.notdef805=21+1455|.notdef293=20+5|.notdef279=19+1116|.notdef233=18+571|.notdef293=17+5|.notdef276=16+1228|.notdef18694=13+1032|.notdef225=12+299|.notdef1651=10+0|.notdef457=10@-270,420+0|.notdef1547=10+1375|.notdef293=9+5|.notdef247=8+1169|.notdef18504=6+1251|.notdef271=5+591|.notdef293=4+5|.notdef280=3+1873|.notdef16651=0+1619]

#Highlighting the only difference via `git diff -U0 --word-diff --word-diff-regex=. --no-index -- <
.notdef457=10@[-60-]{+-27+}0,[-75-]{+420+}+0
```

**Comment by avih** [comment]
> this appears to be a bug in HarfBuzz

It might be the case, but I don't think there's enough evidence so far to suggest that.

What @TheOneric suggested is presumably invalidated by the following comment by @TheBluuDot as far as I can tell.

**Comment by avih** [comment]
> So I checked the previous versions and this issue popped up somewhere between [fd63bf3](https://github.com/mpv-player/mpv/commit/fd63bf398af14c95ee7df18815dc3c9d1b4d9b49) and [d92cf77](https://github.com/mpv-player/mpv/commit/d92cf77be56190dfd7fbac3724f290ceb6ff9437).

Can you please recheck these commits?

Because for me both these builds are OK:
- mpv-x86_64-20211219-git-fd63bf3
- mpv-x86_64-20211226-git-d92cf77

And this build has the issue:
- mpv-x86_64-20220109-git-777628e

And this is in line with the earlier comment which points at [the harfbuzz commit ](https://github.com/harfbuzz/harfbuzz/commit/43be5ba442548528c89ad31c0927cc68515b736e) from 2022-01-04.

**Comment by TheBluuDot** [comment]
Ah I rechecked and I seem to have made a mistake. The issue did indeed pop up in the `mpv-x86_64-20220109-git-777628e` build. In that case, I should file an issue with HarfBuzz pointing to @TheOneric's comment since I could not replicate the results on my end. Thanks a lot for rechecking and I apologize for the inconvenience.

**Comment by ValZapod** [comment]
Well, it is fixed now.

**Comment by TheBluuDot** [comment]
Yeah it was fixed by harfbuzz/harfbuzz#3489.
